src/account_manager/operations/stronghold_backup/mod.rs

```diff
@@ -0,0 +1,178 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{fs, path::PathBuf, sync::atomic::Ordering};
+
+use iota_client::secret::{stronghold::StrongholdSecretManager, SecretManager, SecretManagerDto};
+use zeroize::Zeroize;
+
+use crate::account_manager::{AccountHandle, AccountManager, AccountManagerBuilder};
+
+mod stronghold_snapshot;
+use stronghold_snapshot::{read_data_from_stronghold_snapshot, store_data_to_stronghold};
+
+impl AccountManager {
+    /// Backup the account manager data in a Stronghold file
+    /// stronghold_password must be the current one when Stronghold is used as SecretManager.
+    pub async fn backup(&self, backup_path: PathBuf, mut stronghold_password: String) -> crate::Result<()> {
+        log::debug!("[backup] creating a stronghold backup");
+        let mut secret_manager = self.secret_manager.write().await;
+
+        let secret_manager_dto = SecretManagerDto::from(&*secret_manager);
+
+        match &mut *secret_manager {
+            // Backup with existing stronghold
+            SecretManager::Stronghold(stronghold) => {
+                stronghold.set_password(&stronghold_password).await?;
+
+                store_data_to_stronghold(self, stronghold, secret_manager_dto).await?;
+
+                // Get current snapshot_path to set it again after the backup
+                let current_snapshot_path = stronghold.snapshot_path.clone();
+
+                // Set the snapshot path
+                stronghold.snapshot_path = backup_path;
+
+                // Write snapshot to backup path
+                stronghold.write_stronghold_snapshot().await?;
```

Comment:
Having to set/reset the path is a bit of a weird API. Maybe an alternative would be to have `write_stronghold_snapshot` take an Option, with None it would use its own path and with Some it would use the given one. Just a thought, do whatever you want with it :)

Reply:
Alright, I can do that later if you want